src/Language/AST/NodeList.php

```diff
@@ -91,17 +91,15 @@ public function offsetGet($offset)// : Node
     public function offsetSet($offset, $value) : void
     {
         if (is_array($value)) {
-            if (isset($value['kind'])) {
-                /** @phpstan-var T $node */
-                $node                 = AST::fromArray($value);
-                $this->nodes[$offset] = $node;
+            /** @phpstan-var T $value */
+            $value = AST::fromArray($value);
```

Comment:
wouldn't it be better to keep var name as $node so we don't change type of `$value argument` to `T`?

Reply:
Well, we *do* change its type. Since it was `array|T` before and headed into this branch because it `is_array()`, we now successfully narrowed it.

Reply:
aight 👍🏾